File: backend/app/schemas/food.py
from __future__ import annotations

from datetime import date

from pydantic import BaseModel, Field

from app.schemas.common import SourceAttribution


class FoodEstablishmentSearchResult(BaseModel):
    id: str
    fhrs_id: str
    business_name: str
    business_type: str | None = None
    address: str | None = None
    postcode: str | None = None
    rating_value: str | None = None
    rating_date: date | None = None
    local_authority_name: str | None = None
    scheme_type: str | None = None
    new_rating_pending: bool | None = None
    source: SourceAttribution


class FoodSearchResponse(BaseModel):
    query: str
    results: list[FoodEstablishmentSearchResult]
    total: int
    dataset_version: str | None = None
    message: str | None = None


class FoodEstablishmentView(FoodEstablishmentSearchResult):
    local_authority_business_id: str | None = None
    rating_key: str | None = None
    hygiene_score: int | None = None
    structural_score: int | None = None
    confidence_in_management_score: int | None = None
    longitude: float | None = None
    latitude: float | None = None

File: frontend/types/food.ts
import type { SourceAttribution } from "./common";

export interface FoodEstablishmentSearchResult {
  id: string;
  fhrs_id: string;
  business_name: string;
  business_type: string | null;
  address: string | null;
  postcode: string | null;
  rating_value: string | null;
  rating_date: string | null;
  local_authority_name: string | null;
  scheme_type: string | null;
  new_rating_pending: boolean | null;
  source: SourceAttribution;
}

export interface FoodSearchResponse {
  query: string;
  results: FoodEstablishmentSearchResult[];
  total: number;
  dataset_version: string | null;
  message: string | null;
  suggestions: FoodEstablishmentSearchResult[];
}

export interface FoodEstablishmentView extends FoodEstablishmentSearchResult {
  local_authority_business_id: string | null;
  rating_key: string | null;
  hygiene_score: number | null;
  structural_score: number | null;
  confidence_in_management_score: number | null;
  longitude: number | null;
  latitude: number | null;
}
